cwl/src/main/scala/cwl/ExpressionEvaluator.scala

```diff
@@ -24,6 +25,16 @@ object ExpressionEvaluator {
   type ECMAScriptFunction = String Refined MatchesRegex[ECMAScriptFunctionWitness.T]
   type MatchesECMAFunction = MatchesRegex[ECMAScriptFunctionWitness.T]
 
+  // This is not ECMAScript, just what CWL uses for interpolated expressions. The pattern is 'before$(expression)after'.
+  // The regex uses a non-greedy quantifier on the 'before' to allow the expression to be processed from left to right,
+  // and there are capturing groups around each portion. The official specification for interpolated strings is in the
+  // last part of this section:
+  // http://www.commonwl.org/v1.0/CommandLineTool.html#Parameter_references
+  val InterpolatedStringWitness = Witness("""(?s)(.*?)\$\(([^\)]+)\)(.*)""")
+  val InterpolatedStringRegex = InterpolatedStringWitness.value.r
+  type InterpolatedString = String Refined MatchesRegex[InterpolatedStringWitness.T]
```

Comment:
ToL could be `String Refined MatchesInterpolatedString` as you declared the type alias below

Reply:
Interesting, I essentially copy/pasted from the `ECMAScript{Function|Expression}` definitions above this and assumed there was a reason why it had to be this way. :man_shrugging: